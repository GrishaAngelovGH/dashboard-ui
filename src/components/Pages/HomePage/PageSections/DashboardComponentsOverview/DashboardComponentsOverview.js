import DashboardOverviewSection from "../DashboardOverviewSection/DashboardOverviewSection"

const DashboardComponentsOverview = () => (
    <div>test</div>
)

export default DashboardComponentsOverview