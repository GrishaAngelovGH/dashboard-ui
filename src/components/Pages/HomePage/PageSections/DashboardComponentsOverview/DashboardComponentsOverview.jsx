import PageSection from 'components/Pages/PageSection'
import DashboardComponent from './DashboardComponent'

const DashboardComponentsOverview = ({ components }) => (
	<PageSection style={{ background: '#ebedef', height: 'auto' }}>
		<div className='row g-0'>
			<div className='col-md-12 mt-2'>
				<div className='row g-0 justify-content-center m-3'>
					<div className='col-md-8 text-center'>
						<h2>Components for Dashboards</h2>
					</div>
				</div>

				<div className='row g-0 justify-content-center mb-4 text-center'>
					<div className='col-md-8 col-11'>
						Using Dashboards, you can easily create informative and understandable panels
						using a wide range of components. All you need is to drag and drop the data
						to the appropriate fields and adjust them using a wide variety of properties.
					</div>
				</div>

				<div className='row g-0 justify-content-center'>
					<div className='col-md-6 col-lg-10 pl-4'>
						{
							components.map((v, i) => (
								<div key={i} className='row g-0 mb-4'>
									{
										v.map(item => (
											<div className='col-md-12 col-lg-4 pb-2' key={item.title}>
												<DashboardComponent
													image={item.image}
													title={item.title}
													description={item.description}
												/>
											</div>
										))
									}
								</div>
							))
						}
					</div>
				</div>
			</div>
		</div>
	</PageSection >
)

export default DashboardComponentsOverview