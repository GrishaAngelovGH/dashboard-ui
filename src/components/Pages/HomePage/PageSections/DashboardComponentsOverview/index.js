import DashboardComponentsOverview from './DashboardComponentsOverview'

export default DashboardComponentsOverview