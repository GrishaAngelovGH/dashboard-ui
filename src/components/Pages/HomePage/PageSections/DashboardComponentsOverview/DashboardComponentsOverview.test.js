import { render } from '@testing-library/react'

import DashboardComponentsOverview from './DashboardComponentsOverview'

test('should render component', () => {

    const components = [
        [
            { image: 'chart', title: 'Chart', description: 'Add various types of graphs for data analysis' },
        ],
        [
            { image: 'pivot', title: 'Pivot', description: 'Quickly summarize large amounts of data in pivot tables' },
        ]
    ]

    const container = render(<DashboardComponentsOverview components={components} />)

    expect(container).toMatchSnapshot()
})