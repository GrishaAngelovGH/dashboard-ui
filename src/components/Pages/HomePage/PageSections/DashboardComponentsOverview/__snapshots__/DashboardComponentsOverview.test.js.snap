// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row no-gutters"
        style="background: rgb(235, 237, 239); height: auto;"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-12 mt-2"
            >
              <div
                class="row no-gutters justify-content-center m-3"
              >
                <div
                  class="col-md-8 text-center"
                >
                  <h2>
                    Components for Dashboards
                  </h2>
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-4 text-center"
              >
                <div
                  class="col-md-8 col-11"
                >
                  Using Dashboards, you can easily create informative and understandable panels using a wide range of components. All you need is to drag and drop the data to the appropriate fields and adjust them using a wide variety of properties.
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center"
              >
                <div
                  class="col-md-6 col-lg-10 pl-4"
                >
                  <div
                    class="row no-gutters mb-4"
                  >
                    <div
                      class="col-md-12 col-lg-4 pb-2"
                    >
                      <div
                        class="dashboard-component text-center"
                      >
                        <img
                          height="300"
                          src="chart.png"
                          width="310"
                        />
                        <h2>
                          Chart
                        </h2>
                        <p>
                          Add various types of graphs for data analysis
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="row no-gutters mb-4"
                  >
                    <div
                      class="col-md-12 col-lg-4 pb-2"
                    >
                      <div
                        class="dashboard-component text-center"
                      >
                        <img
                          height="300"
                          src="pivot.png"
                          width="310"
                        />
                        <h2>
                          Pivot
                        </h2>
                        <p>
                          Quickly summarize large amounts of data in pivot tables
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row no-gutters"
      style="background: rgb(235, 237, 239); height: auto;"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-12 mt-2"
          >
            <div
              class="row no-gutters justify-content-center m-3"
            >
              <div
                class="col-md-8 text-center"
              >
                <h2>
                  Components for Dashboards
                </h2>
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-4 text-center"
            >
              <div
                class="col-md-8 col-11"
              >
                Using Dashboards, you can easily create informative and understandable panels using a wide range of components. All you need is to drag and drop the data to the appropriate fields and adjust them using a wide variety of properties.
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center"
            >
              <div
                class="col-md-6 col-lg-10 pl-4"
              >
                <div
                  class="row no-gutters mb-4"
                >
                  <div
                    class="col-md-12 col-lg-4 pb-2"
                  >
                    <div
                      class="dashboard-component text-center"
                    >
                      <img
                        height="300"
                        src="chart.png"
                        width="310"
                      />
                      <h2>
                        Chart
                      </h2>
                      <p>
                        Add various types of graphs for data analysis
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="row no-gutters mb-4"
                >
                  <div
                    class="col-md-12 col-lg-4 pb-2"
                  >
                    <div
                      class="dashboard-component text-center"
                    >
                      <img
                        height="300"
                        src="pivot.png"
                        width="310"
                      />
                      <h2>
                        Pivot
                      </h2>
                      <p>
                        Quickly summarize large amounts of data in pivot tables
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
